src/etc/inc/interfaces.inc

```diff
@@ -3950,6 +3990,17 @@ function interface_configure($interface = "wan", $reloadall = false, $linkupeven
 	if ($wantedmtu == 0 && interface_is_vlan($mtuif) != NULL && interface_isppp_type($interface)) {
 		$wantedmtu = interface_mtu_wanted_for_pppoe($mtuhwif);
 	}
+	if ($wantedmtu == 0 && interface_is_type($mtuif, 'gre')) {
```

Comment:
The first part here needs its own set of parenthesis. I know it matches the block above, but the block above is also wrong and should be fixed. Each test should be isolated in parenthesis.

Reply:
fixed